### DISCUSSION TITLE
Trying to compile mpi4py-4.0.0 form source in a drake site

### DISCUSSION THREAD

**Discussion by asfbar** [mpi4py/mpi4py#563] [Q&A]
Using anaconda3 (python 3.11.7) 
gcc 9.4.0
openMpi 4.1.1
Cython 3.0.0a11

**while executing:**
`**python setup.py build --mpicc=/path/to/mpicc**`

**got the error:**
```
Error compiling Cython file:
you may be using mpiexec or mpirun
from a different MPI implementation
envvar,getenv(envvar), vendor,
....
```

can you please advise?
Thank you.

**Comment by dalcinl** [comment]
As you are posting partial information, I cannot really advise. There seem to be an error coming from Cython, but you do not show what that error is, by the way, using Cython 3.0.0a11 (an old alpha pre release version) is a bit weird, why are you doing that? 
It also looks like you have multiple MPI installed, with mpiexec not matching what was used to build mpi4py. However, if Cython failed with error, then how come you got to the point of getting that message about using different MPI implementations??
Also, you did not mention what mpi4py version you are trying to build. Also, why are you invoking setup.py directly rather than using `pip install ...`?
